.github/workflows/run-tests.yml

```
@@ -100,6 +105,42 @@ jobs:
         run: |
           cat "$GITHUB_OUTPUT"
 
+      - name: Checkout Code
+        uses: actions/checkout@v4
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          ref: ${{ env.COMMIT }}
+          fetch-depth: 0
+
+      - name: Fetch base branch
+        run: git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }}
+
+      - name: Compute merge base
+        run: |
+          MERGE_BASE="$(git merge-base "${{ env.COMMIT }}" "${{ github.event.pull_request.base.ref }}")"
+          echo "MERGE_BASE=${MERGE_BASE}" >> "$GITHUB_ENV"
+
+      - name: Collect Modified Tests
+        id: collect_modified_tests
+        run: |
+          set -euo pipefail
+
+          echo "MERGE_BASE=${MERGE_BASE} COMMIT=${COMMIT}"
+
+          OUTPUT=$(go run ./cmd/tools/test/find_altered_tests.go \
+            -c unit \
+            -c integration \
+            -c functional \
+            -c functional_ndc \
+            -c functional_xdc \
+            -s "${MERGE_BASE}" \
+            -t "${COMMIT}")
+
+          echo "OUTPUT: $OUTPUT"
+
+          echo "$OUTPUT" | grep -E '^(modified_unit_test_suites|modified_integration_test_suites|modified_functional_test_suites|modified_functional_ndc_test_suites|modified_functional_xdc_test_suites)=' >> "$GITHUB_ENV"
```

```suggestion
          go run ./cmd/tools/test/find_altered_tests.go \
            -c unit \
            -c integration \
            -c functional \
            -c functional_ndc \
            -c functional_xdc \
            -s "${MERGE_BASE}" \
            -t "${COMMIT}" | tee -a "$GITHUB_ENV"
```

And make sure your Go script only produces relevant output to stdout and logs to stderr.